sim/midas/src/main/scala/midas/models/dram/IngressUnit.scala

```diff
@@ -94,28 +106,60 @@ class IngressModule(val cfg: BaseConfig)(implicit val p: Parameters) extends Mod
                        ((awCredits.value < wCredits.value) && awCredits.inc) ||
                         awCredits.inc && wCredits.inc
 
+  val read_req_done = arQueue.io.enq.fire()
   when (!io.relaxed) {
     Seq(awCredits, wCredits) foreach { _.dec := write_req_done }
   }
 
+  val readMatcher = Module(new FIFOAddressMatcher(cfg.maxReads*2, addrWidth)).io
+  readMatcher.enq.valid := !ingressUnitStall && io.nastiInputs.hBits.ar.valid
```

Comment:
replace !ingressUnitStall with tFireHelper.fire(readMatcher.enq.ready), once you've added it to the list.